packages/driver/src/cli.mts

```diff
@@ -0,0 +1,290 @@
+    (await installEdgeDbCli()) ??
+    null;
+
+  if (cliLocation === null) {
+    throw Error("Failed to find or install EdgeDB CLI.");
+  }
+
+  return runEdgeDbCli(args, cliLocation);
+}
+
+async function installEdgeDbCli(): Promise<string> {
+  debug("Installing EdgeDB CLI...");
+  await downloadCliPackage();
+
+  const installDir = getInstallDir(TEMPORARY_CLI_PATH);
+  const binaryPath = path.join(installDir, "edgedb");
+  await fs.writeFile(CLI_LOCATION_CACHE_FILE_PATH, binaryPath, {
+    encoding: "utf8",
+  });
+  debug("CLI installed at:", binaryPath);
+
+  try {
+    await fs.access(binaryPath, fs.constants.F_OK);
+  } catch {
+    debug("Self-installing EdgeDB CLI...");
+    selfInstallEdgeDbCli(TEMPORARY_CLI_PATH);
+  }
+  return binaryPath;
+}
+
+async function downloadCliPackage() {
+  console.log("No EdgeDB CLI found, downloading CLI package...");
+  debug("Downloading CLI package...");
+  const cliPkg = await findPackage();
+  const downloadDir = path.dirname(TEMPORARY_CLI_PATH);
+  await fs.mkdir(downloadDir, { recursive: true }).catch((error) => {
+    if (error.code !== "EEXIST") throw error;
+  });
+  const downloadUrl = new URL(cliPkg.installref, EDGEDB_PKG_ROOT);
+  await downloadFile(downloadUrl, TEMPORARY_CLI_PATH);
+  debug("CLI package downloaded to:", TEMPORARY_CLI_PATH);
+
+  await fs.chmod(TEMPORARY_CLI_PATH, 0o755);
+  const stat = await fs.stat(TEMPORARY_CLI_PATH);
+  debug("CLI package stats:", stat);
+}
+
+async function getCliLocationFromCache(): Promise<string | null> {
+  debug("Checking CLI cache...");
+  try {
+    const cachedBinaryPath = (
+      await fs.readFile(CLI_LOCATION_CACHE_FILE_PATH, { encoding: "utf8" })
+    ).trim();
+      debug("CLI path in cache at:", cachedBinaryPath);
+    try {
+      await fs.access(cachedBinaryPath, fs.constants.F_OK);
+      debug("CLI binary found in path:", cachedBinaryPath);
+      return cachedBinaryPath;
+    } catch (err) {
+      // EdgeDB binary not found in the cached directory
+      debug("No CLI found in cache.", err);
+      return null;
+    }
+  } catch (err) {
+    // Cache file does not exist or other error
+    debug("Cache directory does not exist.", err);
+    return null;
+  }
+}
+
+async function whichEdgeDbCli() {
+  debug("Checking if CLI is in PATH...");
+  const location = await which("edgedb", { nothrow: true });
+  debug(`CLI found in PATH at: ${location}`);
+  if (location) {
+    return location;
+  }
+  return null;
+}
+
+function runEdgeDbCli(
+  args: string[],
+  pathToCli: string | null,
+  execOptions: ExecSyncOptions = { stdio: "inherit" }
+) {
+  const cliCommand = pathToCli ?? "edgedb";
+  const command = `${cliCommand} ${args.join(" ")}`;
+  debug(`Running EdgeDB CLI: ${command}`);
+  return execSync(command, execOptions);
+}
+
+function selfInstallEdgeDbCli(pathToCli: string) {
+  debug("Self-installing EdgeDB CLI...");
+  return runEdgeDbCli(["_self_install"], pathToCli);
+}
+
+async function findPackage(): Promise<Package> {
```

Comment:
All of this logic is adapted from https://github.com/edgedb/setup-edgedb/blob/main/src/main.ts#L123 and ought to be centralized into a separate package to keep them in sync with each other, but there are some slight differences between the GitHub Actions runtime and a plain Node runtime, so I decided the easiest thing to do for now was to copy.

We can still look at making a common abstraction they can share (probably this logic without the side effects), but I don't want to block this effort on that.